predictions_FB.py

```diff
@@ -0,0 +1,92 @@
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+from fbprophet import Prophet
+import yfinance as yf
+import math
+import os
+import sys
+import pandas as pd
+pd.options.mode.chained_assignment = None
+
+
+class suppress_stdout_stderr(object):
+    '''
+    Used to supress the verbose output of the inner workings of the FB prophet model
+    '''
+    def __init__(self):
+        self.null_fds = [os.open(os.devnull, os.O_RDWR) for x in range(2)]
+        self.save_fds = (os.dup(1), os.dup(2))
+
+    def __enter__(self):
+        os.dup2(self.null_fds[0], 1)
+        os.dup2(self.null_fds[1], 2)
+
+    def __exit__(self, *_):
+        os.dup2(self.save_fds[0], 1)
+        os.dup2(self.save_fds[1], 2)
+        os.close(self.null_fds[0])
+        os.close(self.null_fds[1])
+
+def fb_predict(dfi, plot = False, days =1):
+    '''
+    This function takes in:
+    dfi - A dataframe with just the column of close price named as Close and date as index 
+    plot - Whether you want to plot the resulyts or not( Default is False)
+    days - Number of days into the future you want to predict (Dafault is 1)
+
+    This function returns:
+    mean_err - The mean percentage error in calculations
+    pred - A pandas datafrmae with predictions, upper and lower bounds
+
+    '''
+    df = dfi.copy()
+    arr = [i for i in range(len(df))]
+    ar = df.index
+    df.index= arr
+    df['ds'] = ar
+    df = df[['ds', 'Close']]
+    df.rename(columns = {"Close":'y'}, inplace = True)
+
+    train_length = math.floor(0.1*len(df))
+    train = df[:-train_length]
+    test = df[-train_length:]
+    m = Prophet(daily_seasonality = True) 
+    with suppress_stdout_stderr():
+        m.fit(train) 
+    future = m.make_future_dataframe(periods=train_length) 
+    prediction = m.predict(future)
+    test_predict = prediction[-train_length:]
+    test['predict'] = test_predict['trend']
+    test['error'] =((test['y'] - test['predict'])/test['y'])*100
+    mean_err = round(test['error'].mean(),3)
```

Comment:
Do we want to keep the test models in production?

Reply:
This has been implemented to calculate the error by splitting the data into test and train as the main model trains on the entire historic data. If we have some other workaround then I would love to remove the extra model.

Reply:
If we don't need this in the final thing, I'd say we remove this. If you want it in development, I'd recommend having the test model in another file which runs within the `fb_predict` method.  Then you could add that file to the `.gitignore` so that it isn't committed.

Reply:
Or we could even leave it be (in another file or method) and have another flag (in the method signature) to take a boolean param for calculating the error so that the test model isn't run every time.

Reply:
Ok yes I can do that

Reply:
I will make it False by default so if nothing is mentioned, it doesn't run